### ISSUE TITLE
Cannot seek to last frame of video (OSX editor / AVPro 3)

### ISSUE THREAD

**Issue by acgourley** [RenderHeads/UnityPlugin-AVProVideo#1837]
The last frame of video is never reached. Seeking to it, or the frame before it, fails. 

 - Unity version: 2021.3 & 2022.3
 - AVPro Video version 3.0.0
 - Operating system version: OSX 14.2
 - Device model: Mac Studio
 - Video specs: UHD 60

**To Reproduce**
1. Load this test video in a media player https://www.dropbox.com/scl/fi/rep811zhqlbffw71vtjg8/100FrameTestVideo.mp4?rlkey=8idrlw32m512pfv12esas23pf&dl=0 
2. This video has 100 frames, labeled 0-99. 
3. You can play the video and observe it stops on frame 98. If you use the seek <> buttons in the media player editor UI, backwards works, but forward will not. Trying to seek to frame 98 or 99 will also never succeed, according to mediaPlayer.TextureProducer.GetTextureTimeStamp()

Here's a short screencast where I show it happening: https://capture.dropbox.com/rcu4MYsGFLhJ9cXC

**Comment by acgourley** [comment]
Same issue with 2.9.3 it seems.

**Comment by Ste-RH** [comment]
We checked the behavior against Quicktime on macOS and it is identical to AVPro Video.  Also, every platform AVPro Video supports appears to behave the same.

We think there is a problem relating to PTS/DTS in your video.

In some tools it reports back as being a framerate of 54.546.  When I try to extract frames to PNGs (using ffmpeg) I get 99 frames output (last one being the one with 98 rendered in it).  So I tried a simple re-encode (ffmpeg.exe -i 100FrameTestVideo.mp4 -c copy -r 59.94 100FrameTestVideo_new.mp4) to affirm the framerate.  This appeared to allow selection/seek of frame 100 (with 99 rendered in it), though in some tools it reports as framerate of 60.245 !

Here is the video:

[100FrameTestVideo_new.zip](https://github.com/user-attachments/files/17434627/100FrameTestVideo_new.zip)

Bottom line here.  We think there is an issue with this video and you need to re-encode it.
